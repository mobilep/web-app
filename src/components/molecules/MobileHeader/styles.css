.MobileHeader {
	width: 100%;
	padding: 16px 0;
	display: flex;
	align-items: center;
	justify-content: space-between;
	height: var(--mobile-header-height);
}

.MobileHeader-closeButton {
	cursor: pointer;
	width: 31px;
	height: 31px;
	border: none;
	padding: 0;
	background: none;
	color: var(--tomatoRed);
}

.MobileHeader-closeArrowButton {
	width: 25px;
	height: 25px;
	display: flex;
}

.MobileHeader-closeArrowButton svg {
	transform: rotate(180deg);
	width: 15px;
	height: 15px;
}

.MobileHeader-title {
	display: flex;
	justify-content: center;
	align-items: center;
	flex-direction: column;
	font-size: 20px;
	font-weight: 600;
	white-space: nowrap;
}

.MobileHeader-title p {
	margin-bottom: 0;
	color: var(--grey4);
	font-size: 12px;
}

.MobileHeader-buttons:empty {
	width: 25px;
}

@media (max-width: 768px) {
	.MobileHeader {
		position: relative;
	}

	.MobileHeader-title {
		font-size: 16px;
		padding-right: 10px;
		position: absolute;
		left: 50%;
		transform: translateX(-50%);
	}
}
