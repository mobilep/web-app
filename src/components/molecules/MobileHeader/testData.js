export default {
	buttonsProps: [
		{
			label: 'Edit',
			onClick: () => {},
		},
		{
			label: 'Delete',
			onClick: () => {},
		},
	],
	title: 'Mobile header',
};
