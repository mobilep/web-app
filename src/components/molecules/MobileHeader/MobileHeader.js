import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { HeaderDropdown } from '..';
import { Button } from '../../atoms';
import { CloseIcon, ArrowIcon } from '../../../icons';
import './styles.css';

export default class MobileHeader extends Component {

	static propTypes = {
		arrowClassName: PropTypes.string,
		buttonsProps: PropTypes.arrayOf(PropTypes.shape({
			label: PropTypes.string.isRequired,
			onClick: PropTypes.func.isRequired,
			disabled: PropTypes.bool,
		})),
		className: PropTypes.string,
		closeButtonHandler: PropTypes.func,
		crossIconType: PropTypes.bool,
		description: PropTypes.bool,
		descriptionText: PropTypes.string,
		editMode: PropTypes.bool,
		hasNoButton: PropTypes.bool,
		hasNoMenu: PropTypes.bool,
		menuClassName: PropTypes.string,
		title: PropTypes.string,
		updateDate: PropTypes.string,
	}

	getClassName ({ className } = this.props) {
		return classNames('MobileHeader', className);
	}

	getMenuClassName = (className) => classNames('MobileHeader-buttons', className);

	getButtons ({ buttonsProps } = this.props) {
		if (!buttonsProps) return null;
		if (buttonsProps.length > 1) {
			return <HeaderDropdown buttonsProps={buttonsProps} />;
		} else if (buttonsProps[0].label === 'Send') {
			return;
		} else {
			const { onClick, label, disabled } = buttonsProps[0];
			return (
				<Button className='Button-text p-r-0' disabled={disabled} onClick={onClick}>
					{label}
				</Button>
			);
		}

	}

	getArrowButtonClassName = (className) => classNames('MobileHeader-closeArrowButton', className);

	render () {
		const {
			crossIconType,
			closeButtonHandler,
			title,
			hasNoButton,
			menuClassName,
			hasNoMenu,
			arrowClassName,
			description,
			descriptionText,
		} = this.props;

		const closeArrowButtonClass = this.getArrowButtonClassName(arrowClassName);
		const closeButtonClass = `MobileHeader-closeButton ${crossIconType ? '' : closeArrowButtonClass}`;
		const className = this.getClassName();
		return (
			<div className={className}>
				{!hasNoButton &&
					<button
						className={closeButtonClass}
						onClick={closeButtonHandler}
					>
						{crossIconType
							? <CloseIcon />
							: <ArrowIcon />
						}
					</button>}
				<div className='MobileHeader-title'>{title}
					{description && <p>{descriptionText}</p>}
				</div>
				{!hasNoMenu &&
					<div className={this.getMenuClassName(menuClassName)}>{this.getButtons()}</div>}
			</div>
		);
	}

}
