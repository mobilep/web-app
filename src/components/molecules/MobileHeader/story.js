import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs/react';

import MobileHeader from './MobileHeader';
import MobileHeaderProps from './testData';

const props = () => (
	{
		title: text('Title', MobileHeaderProps.title),
		buttonsProps: MobileHeaderProps.buttonsProps,
		crossIconType: boolean('Cross Icon Type', false),
	}
);

storiesOf('2. Molecules / MobileHeader', module)
	.add('MobileHeader', () => (
		<MobileHeader {...props()} />
	));
