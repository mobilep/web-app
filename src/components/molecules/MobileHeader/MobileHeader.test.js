import React from 'react';
import { shallow } from 'enzyme';

import MobileHeader from './MobileHeader';

import MobileHeaderProps from './testData';

describe('Molecule - MobileHeader', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<MobileHeader {...MobileHeaderProps} />
		);
		expect(wrapper).toMatchSnapshot();
	});

});
