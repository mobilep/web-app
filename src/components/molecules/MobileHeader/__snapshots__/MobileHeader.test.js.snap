// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - MobileHeader Should render 1`] = `
<div
  className="MobileHeader"
>
  <button
    className="MobileHeader-closeButton MobileHeader-closeArrowButton"
  >
    <ArrowIcon />
  </button>
  <div
    className="MobileHeader-title"
  >
    Mobile header
  </div>
  <div
    className="MobileHeader-buttons"
  >
    <HeaderDropdown
      buttonsProps={
        Array [
          Object {
            "label": "Edit",
            "onClick": [Function],
          },
          Object {
            "label": "Delete",
            "onClick": [Function],
          },
        ]
      }
    />
  </div>
</div>
`;
